source/common/router/config_impl.h

```diff
@@ -1234,6 +1234,7 @@ class RouteMatcher {
 
   VirtualHostImplSharedPtr default_virtual_host_;
   const bool ignore_port_in_host_matching_{false};
+  const std::string vhost_header_;
```

Comment:
```suggestion
  const Http::LowerCaseString vhost_header_;
```

to avoid create the lower case string copy for every match.